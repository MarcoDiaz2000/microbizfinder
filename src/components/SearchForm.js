import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { FiSearch } from 'react-icons/fi';

const SearchForm = () => {
  const [term, setTerm] = useState('');
  const navigate = useNavigate();

  const handleSubmit = (e) => {
    e.preventDefault();

    navigate(`/services/${term}`);
  };

  return (
    <form onSubmit={handleSubmit} className="search-form">
      <label htmlFor="what">
        <input
          type="text"
          value={term}
          onChange={(e) => setTerm(e.target.value)}
          placeholder="Ex: food, drink, shopping, etc."
          id="what"
          className="search-input"
        />
      </label>

      <button type="submit" className="search-button">
        <FiSearch size={20} color="white" />
        <span>Search</span>
      </button>
    </form>
  );
};

export default SearchForm;
