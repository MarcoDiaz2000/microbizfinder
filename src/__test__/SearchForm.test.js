import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { useNavigate } from 'react-router-dom';
import SearchForm from '../components/SearchForm';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('SearchForm', () => {
  it('navigates to the correct path on form submission', () => {
    const navigate = jest.fn();
    useNavigate.mockReturnValue(navigate);

    const { getByPlaceholderText, getByText, asFragment } = render(<SearchForm />);

    const input = getByPlaceholderText('Ex: food, drink, shopping, etc.');
    const submitButton = getByText('Search');

    const searchTerm = 'food';
    fireEvent.change(input, { target: { value: searchTerm } });
    fireEvent.click(submitButton);

    expect(navigate).toHaveBeenCalledWith('/services/food');
    expect(asFragment()).toMatchSnapshot();
  });
});
