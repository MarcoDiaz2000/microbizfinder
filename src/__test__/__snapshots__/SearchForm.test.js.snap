// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm navigates to the correct path on form submission 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <label
      for="what"
    >
      <input
        class="search-input"
        id="what"
        placeholder="Ex: food, drink, shopping, etc."
        type="text"
        value="food"
      />
    </label>
    <button
      class="search-button"
      type="submit"
    >
      <svg
        color="white"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: white;"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <span>
        Search
      </span>
    </button>
  </form>
</DocumentFragment>
`;
